crates/core/src/db/commit_log.rs

```diff
@@ -359,6 +469,107 @@ impl From<message_log::Segments> for Iter {
     }
 }
 
+/// Iterator created by [`CommitLog::replay`].
+///
+/// Similar to [`Iter`], but performs integrity checking and maintains
+/// additional state.
+#[must_use = "iterators are lazy and do nothing unless consumed"]
+struct Replay {
```

Comment:
It would be nice to replace `Iter` with this, so the error detection is available by default. However, `Iter` is supposed to be able to start iteration at a specified commit offset, so this is not currently possible.